test/gtest/CMakeLists.txt

```diff
@@ -58,10 +58,29 @@ function(add_gtest TEST_NAME TEST_CPP)
     "ENVIRONMENT;MIOPEN_DEBUG_TUNING_ITERATIONS_MAX=${MIOPEN_DEBUG_TUNING_ITERATIONS_MAX};"
     "ENVIRONMENT;MIOPEN_TEST_WITH_MIOPENDRIVER=${MIOPEN_TEST_WITH_MIOPENDRIVER};")
 
+    set(MIOPEN_TEST_PREFIX)
+
+    if (MIOPEN_TEST_ALL)
+      set(MIOPEN_TEST_PREFIX Full)
+    else()
+      set(MIOPEN_TEST_PREFIX Smoke)
+    endif()
```

Comment:
I think MIOPEN_TEST_ALL used to run all tests (including the smoke tests), but now it will only test things that aren't marked as smoke tests. 

At least when I look at the tests that were previously smoke tests, it looks like they just didn't check the MIOPEN_TEST_ALL env variable, and would be included in both runs.

Reply:
It depends on a set of nodes.
If we run "TEST_ALL" and "not TEST_ALL" stages on the same set of nodes, then it doesn't make any sense at all.
If "TEST_ALL" uses wider set of nodes (or "not TEST_ALL" stages choose smaller available subset of the same nodes), probably we can reconsider and include smoke tests back into "ALL" for now, but in the future we should consider proper job splitting.

The problem is - we are running the same set of tests on the same nodes multiple times.

Reply:
Yea I agree, and I think we should have a discussion about the smoke stage.
I think it would be fine to remove running smoke during the full stages, and make them exclusive sets, but just wanted to make sure it was an intentional change that was discussed.

Reply:
Previously we had a set of binaries that were run from cmake. They checked the "--all" option. If it was there, full tests were run, if not - smoke tests. This option is set if the MIOPEN_TEST_ALL is set to 1. So previously we would run either **full tests** or **smoke tests** (Many tests that were added when switching to Google Test are not very well made; they do not check anything, including these variables) So it needs to be returned to how it was originally.

Reply:
It doesn't matter how the tests treats `--all`/`MIOPEN_TEST_ALL`, the only thing matters is the set of nodes.
If the set of nodes is not the same, it just means that old tests were badly designed.
If the set of nodes is the same, it means the option can be mutually exclusive. That's it.

Mutually exclusive option requires some effort and guarantees from our current CI pipelines. That's out of the scope of this PR.